### PR TITLE
Updated moveitjoy_module.py to add PS3 dualshock model

### PR DESCRIPTION
**Pull request by job-1994**
### Description

Added the PS3 Dualshock controller as was explained [here](https://answers.ros.org/question/325567/unknown-joystick-exception-for-teleop-in-moveit/). Did this because I am looking to control a UR robot jog wise through a PS3 controller and noticed this small fix was not included.

### CODE REVIEW

File: moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py [hunk 1]
```diff
@@ -666,6 +722,8 @@ def waitForInitialPose(self, next_topic, timeout=None):
                 self.marker_lock.release()
 
     def joyCB(self, msg):
+        axes_amount = len(msg.axes)
+        buttons_amount = len(msg.buttons)
```
Comment (rhaschke) [inline moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py:726]: Either use these variables everywhere _instead of_ `len(msg.*)` or ...

File: moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py [hunk 2]
```diff
@@ -676,8 +734,12 @@ def joyCB(self, msg):
             status = PS4Status(msg)
         elif len(msg.axes) == 8 and len(msg.buttons) == 13:
             status = PS4WiredStatus(msg)
+        elif len(msg.axes) == 6 and len(msg.buttons) == 17:
+            status = PS3DualShockStatus(msg)
         else:
-            raise Exception("Unknown joystick")
+            raise Exception(
+                f"Unknown joystick, axes: {axes_amount}, buttons: {buttons_amount}"
```
Comment (rhaschke) [inline moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py:741]: ... use `len(msg.*)` here as well.

File: moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py [hunk 3]
```diff
@@ -169,6 +169,62 @@ def __init__(self, msg):
...
 
 
+class PS3DualShockStatus(JoyStatus):
+    def __init__(self, msg):
+        JoyStatus.__init__(self)
+        # creating from sensor_msgs/Joy
+        if msg.buttons[0] == 1:
+            self.cross = True
+        else:
+            self.cross = False
```
Comment (rhaschke) [inline moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py:179]: Personally, I would prefer the more compact format:
```suggestion
        self.cross = (msg.buttons[0] == 1)
```
However, that could be part of a separate cleanup commit.

**Review by rhaschke — APPROVED** [review]
Thanks. Please consider my nitpicks.

File: moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py (hunk 1 above)
Comment (job-1994) [reply moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py:438]: Done

File: moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py (hunk 2 above)
Comment (job-1994) [reply moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py:453]: Opted for the variable method

**Comment by job-1994** [comment]
The required CI / noetic-ci-shadow-fixed + catkin_lint + ikfast + clang-tidy (delta) (pull_request)  pipeline is failing, but I dont think it is related to the changes ive made.

**Review by rhaschke — COMMENTED** [review]
Thanks for the cleanup. Note that the joy test fails now in [`master-ci`](https://github.com/ros-planning/moveit/actions/runs/1695134720/attempts/1). As it succeeds with `noetic-ci`, I suspect some python2/python3 incompatibility (`master-ci` is running ROS Melodic, `noetic-ci` obviously Noetic). Could you have another look, please?
The failure of `noetic-ci-shadow-fixed + catkin_lint + ikfast + clang-tidy (delta)` is indeed unrelated and should be fixed via 7da00c19b.

**Pull request merged**